src/core.js

```diff
@@ -759,12 +759,7 @@ JSONEditor.prototype = {
           // Remove the type property if it's empty
             delete extended.type
           }
-        } else if (typeof val === 'object' && Array.isArray(val)) {
-          // All other arrays should be intersected (enum, etc.)
-          extended[prop] = val.filter(function (n) {
-            return obj2[prop].indexOf(n) !== -1
-          })
-        } else if (typeof val === 'object' && val !== null) {
+        } else if (typeof val === 'object' && !Array.isArray(val) && val !== null) {
         // Objects should be recursively merged
```

Comment:
The merging of two Object-based Array(like `oneOf`) it will result an empty Array, also in my opinion, use one of the the two array will be better than intersecting them